iscript/src/iscript/macvpn.py

```diff
@@ -0,0 +1,191 @@
+
+    # Assuming we only need to unlock and update the keychain once
+    await unlock_keychain(sign_config["signing_keychain"], sign_config["keychain_password"])
+    await update_keychain_search_path(config, sign_config["signing_keychain"])
+
+    work_dir = config["work_dir"]
+    submodule_dir = os.path.join(top_app.parent_dir, "Mozilla VPN.app", "Contents/Library/")
+
+    # LoginItems inner app
+    loginitems_app = App(
+        orig_path=os.path.join(submodule_dir, "LoginItems/MozillaVPNLoginItem.app"),
+        parent_dir=os.path.join(submodule_dir, "LoginItems"), # Using main app as reference
+        app_path=os.path.join(submodule_dir, "LoginItems/MozillaVPNLoginItem.app"),
+        app_name="MozillaVPNLoginItem.app",
+        formats=task["payload"]["upstreamArtifacts"][0]["formats"],
+        artifact_prefix="public/",
+    )
+    await _sign_app(
+        config, task, sign_config, loginitems_app,
+        entitlements_url=task["payload"]["loginitems-entitlements-url"],
+        provisioning_profile_url=LOGINITEM_PROVISIONING_PROFILE
+    )
+
+    # NativeMessaging inner app
+    nativemessaging_app = App(
+        orig_path=os.path.join(submodule_dir, "NativeMessaging/MozillaVPNNativeMessaging.app"),
+        parent_dir=os.path.join(submodule_dir, "NativeMessaging"), # Using main app as reference
+        app_path=os.path.join(submodule_dir, "NativeMessaging/MozillaVPNNativeMessaging.app"),
+        app_name="MozillaVPNNativeMessaging.app",
+        formats=task["payload"]["upstreamArtifacts"][0]["formats"],
+        artifact_prefix="public/",
+    )
+    await _sign_app(
+        config, task, sign_config, nativemessaging_app,
+        entitlements_url=task["payload"]["nativemessaging-entitlements-url"],
+        provisioning_profile_url=NATIVEMESSAGING_PROVISIONING_PROFILE
+    )
+
+    # Main VPN app
+    # Already defined from extract - just need the extra bits
+    top_app.app_name = "Mozilla VPN.app"
+    top_app.formats = task["payload"]["upstreamArtifacts"][0]["formats"]
+
+    await _sign_app(
+        config, task, sign_config, top_app,
+        entitlements_url=task["payload"]["entitlements-url"],
+        provisioning_profile_url=MOZILLA_VPN_PROVISIONING_PROFILE
+    )
+
+    # Create the PKG and sign it
+    await create_pkg_files(config, sign_config, [top_app])
+
+    # Notarizing only the pkg
+    path_attrs = ["pkg_path"]
+
+    # Need to zip the pkg instead
+    # zip_path = await create_one_notarization_zipfile(config["work_dir"], [app], sign_config, path_attrs=['app_path'])
+    zip_path = os.path.join(config["work_dir"], 'notarization.zip')
+    await _create_notarization_zipfile(config["work_dir"], top_app.pkg_path, zip_path)
+
+    poll_uuids = await notarize_no_sudo(config["work_dir"], sign_config, zip_path)
+    await poll_all_notarization_status(sign_config, poll_uuids)
+
+    log.info("Done notarizing app")
+
+    # TODO: Do we need to staple in place?
```

Comment:
I think there can be some warnings for the user if we don't staple and they launch it without network (or when the notarization service is down) -- best to staple if possible.